Sources/FisherYates/random.swift

```diff
@@ -8,18 +8,31 @@
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 */
 
+#if swift(>=4.0)
+    public typealias RandomInteger = BinaryInteger
+#elseif swift(>=3.0)
+    public typealias RandomInteger = IntegerArithmetic
+#else
+    public typealias RandomInteger = IntegerArithmeticType
+#endif
+
 #if os(OSX)
     import Darwin
 
-    public let random: (Int) -> Int = { Int(arc4random_uniform(UInt32($0))) }
+    public func random<T: RandomInteger> (_ n: T) -> T {
+        numericCast(arc4random_uniform(numericCast(n)))
+    }
 #else
     import Glibc
 
-    public let random: (Int) -> Int = {
+    public func random<T: RandomInteger> (_ n: T) -> T {
+        guard n > 0 else { return 0 }
```

Comment:
I don't think this ever actually falls into an infinite loop–if `n == 0` the code traps due to division by zero, and if `n < 0` `RAND_MAX % numericCast(n)` evaluates to `RAND_MAX % numericCast(-n)`. I'd argue that rather than returning zero in this case, it would be better to just stop with `precondition(_:_:file:line:)` rather than giving a bogus result.